pkg/scm/types.go

```diff
@@ -3,18 +3,49 @@ package scm
 import (
 	"context"
 	"net/http"
+	"strconv"
 	"strings"
 
 	"github.com/jippi/scm-engine/pkg/state"
 	"github.com/jippi/scm-engine/pkg/types"
 )
 
+type Actors []Actor
+
+func (a Actors) Has(actor Actor) bool {
+	for _, item := range a {
+		if item.ID == actor.ID && item.Username == actor.Username {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (a *Actors) Add(actor Actor) {
+	*a = append(*a, actor)
+}
+
 type Actor struct {
+	ID       string
 	Username string
 	Email    *string
 	IsBot    bool
 }
 
+// IntID is a safe parser for the ID field of an Actor, returning -1 if
+// the ID cannot be parsed as an integer.
+func (a Actor) IntID() int {
+	id := strings.Replace(a.ID, "gid://gitlab/User/", "", 1)
```

Comment:
Is there a case where this format could possibly be invalid? (e.g. the last chunk of the string not being a valid user id?)

considering its effectively their PK in the DB I would be fine assuming it would always be valid and we could probably remove the `-1` error logic entirely

Reply:
we could although im wondering if something like `User` would ever be switched out with something else, so rather safe parse instead of letting it panic if ever that time comes